File: Solutions/409. Longest Palindrome/409.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> hm = new HashMap<>();
        int ans = 0;
        int oddAvailable = 0;
        char[] arr = s.toCharArray();
        for(int i=0; i < arr.length; i++){
            hm.put(arr[i], hm.getOrDefault(arr[i], 0) + 1);
        }
        int fre = 0;
        for(Map.Entry<Character, Integer> entrySet : hm.entrySet()){
            fre = entrySet.getValue();
            if(fre % 2 == 0){
                ans += fre;
            }else{
                oddAvailable = 1;
                ans += fre - 1;
            }
        }
        return ans + oddAvailable;
    }
}

File: Solutions/409. Longest Palindrome/409.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        hm = {}
        ans = 0
        is_odd = 0
        for c in s:
            hm[c] = hm.get(c, 0) + 1
        for fre in hm.values():
            if fre % 2 == 0:
                ans += fre
            else:
                is_odd = 1
                ans += fre - 1
        return ans + is_odd